### PR TITLE
[HACK] soundwire: add traces to debug bank switch issues 

### PR DESCRIPTION
**Pull request by bardliao**

### CODE REVIEW

**Comment by plbossart** [comment]
Build is stuck for over 3 hours now. I can't even see it in the internal build results as having started or failed.

**Comment by plbossart** [comment]
SOFCI TEST

**Comment by plbossart** [comment]
Not sure what's going on, now all Linux PRs seem to be building with a Start Time: 2024-06-06 09:16:09 UTC or later

image


Looks like a server restarted @fredoh9 @marc-hb

File: drivers/soundwire/cadence_master.c [hunk 1]
```diff
@@ -929,8 +929,10 @@ irqreturn_t sdw_cdns_irq(int irq, void *dev_id)
 		if (defer && defer->msg) {
 			cdns_fill_msg_resp(cdns, defer->msg,
 					   defer->length, 0);
+			pr_warn("%s defer complete\n", __func__);
 			complete(&defer->complete);
 		} else {
+			pr_warn("%s tx complete\n", __func__);
```
Comment (plbossart) [inline drivers/soundwire/cadence_master.c]: @bardliao this leads to the following useless traces

````
[  259.024629] kernel: sdw_acquire_bus_lock: acquiring locks for stream subdevice #0-Capture
[  259.024642] kernel: sdw_acquire_bus_lock: acquired locks for stream subdevice #0-Capture
[  259.024653] kernel: sdw_transfer
[  259.025011] kernel: sdw_cdns_irq tx complete
[  259.025157] kernel: sdw_transfer
[  259.025352] kernel: sdw_cdns_irq tx complete
[  259.025456] kernel: sdw_transfer
[  259.025634] kernel: sdw_cdns_irq tx complete
[  259.025746] kernel: sdw_transfer
[  259.025920] kernel: sdw_cdns_irq tx complete
[  259.026001] kernel: sdw_transfer
[  259.026127] kernel: sdw_cdns_irq tx complete
[  259.026152] kernel: sdw_transfer
[  259.026387] kernel: sdw_cdns_irq tx complete
[  259.026480] kernel: sdw_transfer
[  259.026586] kernel: sdw_cdns_irq tx complete
[  259.026671] kernel: sdw_transfer
[  259.026926] kernel: sdw_cdns_irq tx complete
[  259.027006] kernel: sdw_transfer
[  259.027105] kernel: sdw_cdns_irq tx complete
[  259.027123] kernel: sdw_transfer
[  259.027207] kernel: sdw_cdns_irq tx complete
[  259.027277] kernel: do_transfer_defer
[  259.027920] kernel: sdw_cdns_irq defer complete
[  259.027945] kernel: sdw_transfer
[  259.028208] kernel: sdw_cdns_irq tx complete
[  259.028243] kernel: sdw_release_bus_lock: releasing locks for stream subdevice #0-Capture
[  259.028247] kernel: sdw_release_bus_lock: released locks for stream subdevice #0-Capture
````

we need to use dev_warn(cdns->dev, "%s defer complete\n", __func__); and dev_warn(cdns->dev, "%s tx complete\n", __func__);

File: drivers/soundwire/bus.c [hunk 2]
```diff
@@ -254,6 +254,7 @@ static inline int do_transfer_defer(struct sdw_bus *bus,
 	enum sdw_command_response resp;
 	int ret = 0, i;
 
+	pr_warn("%s\n", __func__);
```
Comment (plbossart) [inline drivers/soundwire/bus.c]: use bus->dev here.

File: drivers/soundwire/cadence_master.c (hunk 1 above)
Comment (bardliao) [reply drivers/soundwire/cadence_master.c]: fixed

**Comment by bardliao** [comment]
SOFCI TEST

**Comment by bardliao** [comment]
```
[ 1002.420696] sdw_acquire_bus_lock: acquiring locks for stream subdevice #0-Playback
[ 1002.420698] sdw_acquire_bus_lock: acquired locks for stream subdevice #0-Playback
[ 1002.420699] _sdw_prepare_stream: start for stream subdevice #0-Playback
[ 1002.420702] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420756] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420763] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420791] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420794] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420825] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420828] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420857] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420860] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420895] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420897] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420929] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420932] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420957] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420960] soundwire sdw-master-0-2: sdw_transfer
[ 1002.420989] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.420991] soundwire sdw-master-0-2: sdw_transfer
[ 1002.421022] soundwire_intel soundwire_intel.link.2: sdw_cdns_irq tx complete
[ 1002.421034] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421062] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421065] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421094] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421096] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421129] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421132] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421159] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421162] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421198] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421200] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421229] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421232] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421260] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421263] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421295] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421297] soundwire sdw-master-0-3: sdw_transfer
[ 1002.421323] soundwire_intel soundwire_intel.link.3: sdw_cdns_irq tx complete
[ 1002.421333] _sdw_prepare_stream: before bank switch for stream subdevice #0-Playback
[ 1002.421337] soundwire sdw-master-0-2: do_transfer_defer
[ 1002.421344] soundwire sdw-master-0-3: do_transfer_defer
[ 1005.439423] soundwire sdw-master-0-2: Controller Timed out on bank switch
[ 1005.439641] soundwire sdw-master-0-2: multi link bank switch failed: -110
[ 1005.439763] _sdw_prepare_stream: do_bank_switch failed for stream subdevice #0-Playback: -110
[ 1005.439863] sdw_release_bus_lock: releasing locks for stream subdevice #0-Playback
[ 1005.439873] sdw_release_bus_lock: released locks for stream subdevice #0-Playback
[ 1005.439880]  SDW2-Playback: ASoC: error at snd_soc_link_prepare on SDW2-Playback: -110
[ 1005.439913]  SDW2-Playback: ASoC: error at __soc_pcm_prepare on SDW2-Playback: -110
[ 1005.439945]  Speaker: ASoC: error at dpcm_be_dai_prepare on Speaker: -110
[ 1005.439983]  Speaker: ASoC: error at dpcm_fe_dai_prepare on Speaker: -110
[ 1005.440013] sdw_acquire_bus_lock: acquiring locks for stream subdevice #0-Playback
[ 1005.440024] sdw_acquire_bus_lock: acquired locks for stream subdevice #0-Playback
[ 1005.440028] _sdw_prepare_stream: start for stream subdevice #0-Playback
```
Still can't get the reason of "Controller Timed out on bank switch". Just didn't get the CDNS_MCP_INT_RX_WL irq.

**Comment by plbossart** [comment]
well this is progress in my book @bardliao. At least we know it's not the jack interrupt, it's not a race condition between bank switch and other writes, it's squarely a missed interrupt issue.

Do you mind adding a log when we disable the SoundWire interrupts? I don't think it's the reason, because we had multiple transfers active before the bank switch, but it'd prove that the interrupts are on.

**Comment by bardliao** [comment]
SOFCI TEST

**Comment by bardliao** [comment]
SOFCI TEST

**Comment by bardliao** [comment]
SOFCI TEST

**Comment by plbossart** [comment]
Something's not right in the results, I can't see the traces in the logs. 
[dmesg6.log](https://github.com/user-attachments/files/15736973/dmesg6.log)

**Comment by plbossart** [comment]
SOFCI TEST

**Pull request closed without merging**